parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/truth/ConsumerSubject.java

```diff
@@ -0,0 +1,84 @@
+package io.confluent.parallelconsumer.truth;
+
+/*-
+ * Copyright (C) 2020-2022 Confluent, Inc.
+ */
+
+import com.google.common.truth.FailureMetadata;
+import io.confluent.parallelconsumer.model.CommitHistory;
+import io.stubbs.truth.generator.SubjectFactoryMethod;
+import io.stubbs.truth.generator.UserManagedSubject;
+import one.util.streamex.StreamEx;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerParentSubject;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+import javax.annotation.Generated;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static io.confluent.parallelconsumer.truth.CommitHistorySubject.commitHistories;
+
+/**
+ * Optionally move this class into source control, and add your custom assertions here.
+ * <p>
+ * If the system detects this class already exists, it won't attempt to generate a new one. Note that if the base
+ * skeleton of this class ever changes, you won't automatically get it updated.
+ *
+ * @see Consumer
+ * @see ConsumerParentSubject
+ */
+@UserManagedSubject(Consumer.class)
+@Generated(value = "io.stubbs.truth.generator.internal.TruthGenerator", date = "2022-05-17T12:20:38.207945Z")
+public class ConsumerSubject extends ConsumerParentSubject {
+
+    protected ConsumerSubject(FailureMetadata failureMetadata, org.apache.kafka.clients.consumer.Consumer actual) {
+        super(failureMetadata, actual);
+    }
+
+    /**
+     * Returns an assertion builder for a {@link Consumer} class.
+     */
+    @SubjectFactoryMethod
+    public static Factory<ConsumerSubject, Consumer> consumers() {
+        return ConsumerSubject::new;
+    }
+
+    private final Duration timeout = Duration.ofSeconds(10);
+
+    public CommitHistorySubject hasCommittedToPartition(NewTopic topic) {
+        Map<TopicPartition, CommitHistorySubject> map = hasCommittedToPartition(Set.of(topic));
+        return map.values().stream()
+                .findFirst()
+                .orElse(
+                        check("getCommitHistory(%s)", topic.name())
+                                .about(commitHistories())
+                                .that(new CommitHistory(List.of())));
+    }
+
+    public Map<TopicPartition, CommitHistorySubject> hasCommittedToPartition(Set<NewTopic> topic) {
+        Set<TopicPartition> partitions = topic.stream().map(newTopic -> new TopicPartition(newTopic.name(), 0)).collect(Collectors.toSet());
```

Comment:
Rename method to work not with topics but TopicPartitions instead of default partition number